### ISSUE TITLE
Bug - AdminSupportDataCollector

### ISSUE THREAD

**Issue by rkarlsba** [znuny/Znuny#154]
### Environment

- OS:           Linux, RHEL 8.4, x64
- Browser:      Several
- Znuny version: 6.0.36

### Expected behaviour

This work normal

### Actual behaviour

Visit https://myserver.asdf/otrs/index.pl?Action=AdminSupportDataCollector and the httpd/perl process starts traversing through the articles on disk. Since there are 13301654 inodes on /opt/otrs/var, this takes a while. I can't see similar behaviour on OTRS 6.0.30 or on OTRS 5. 

### How to reproduce

I just visit index.pl?Action=AdminSupportDataCollector

It takes 10-30 minutes before the page 

### Additional information

I had several issues when attemting the upgrade and had to go back a few times and start over. I haven't seen this one before. I can't find anything relevant in the logs either.

### Screenshots

Irrelevant

**Comment by rkarlsba** [comment]
PS: To see the process traversing the articles tree, I ran 'strace -fp <thatpid>'

**Comment by hanneshal** [comment]
Hi @rkarlsba 

interesting and thanks for the report. I looked at the plugins and there is no related change. From 6.0.30 to 6.0.36 or newer. The SupportData is collected using this plugins: https://github.com/znuny/Znuny/tree/dev/Kernel/System/SupportDataCollector/Plugin
This method and the list of plugins is almost the same for 5.0 
https://github.com/znuny/Znuny/tree/rel-5_0/Kernel/System/SupportDataCollector/Plugin/

There is no direct check for the article directory (as I can see it right now).
We check if some needed directories are writeable, but the article directory is not one of it. 

https://github.com/znuny/Znuny/blob/dev/Kernel/System/SupportDataCollector/Plugin/OTRS/FileSystemWritable.pm#L33

Then I thought about related API calls like ArticleGet or something like this. And I could find one either. 

The most calls are made against the database and may take some more time. But 10 to 30 minutes is / was never the case.  Mostly because in the default config the Apache would kill the process with a timeout . 

A way to find the issue could be to disable the plugins or plugin folder one by one and test again. 

I would suggest to start with the ones in "OS", starting with disk. Maybe you can pin the one down that creates the high load. 


Regards 
Johannes

**Comment by rkarlsba** [comment]
I found the problem. On the old server, the articles were moved to /home/var because of lack of disk space. When I migrated to the new one, I moved them back to /opt/otrs/var/article. The SupportDataCollector traverses all of /opt/otrs to look for aliens or something, so if the articles are in their default location, it'll traverse them all. i moved them out to /opt/otrs_articles, which solved the problem.
